app/assets/javascripts/modules/repo-description.js

```diff
@@ -18,32 +20,58 @@ moj.Modules.repoDescription = {
   },
 
   bindEvents() {
-    this.$repoSlugSelect.on('change', () => {
-      if (this.$repoSlugSelect.val()) {
-        this.getRepoDescription();
-      } else {
-        this.updateDescription('');
-        $('#repo-results').addClass('js-hidden');
-      }
-    });
     this.$orgSelect.on('change', () => {
       this.showOrgRepos();
     });
   },
 
   showOrgRepos() {
     const currentOrg = this.$orgSelect.val();
+    const repoOptions = Array.from($(`optgroup[label="${currentOrg}"] option`));
+    const repos = repoOptions.map(opt => opt.text);
+    const descriptions = repoOptions.map(opt => opt.dataset.description);
+
+    // reset potentially pre-existing typeahead
+    this.repoSelected = false;
+    this.$repoSlugSelect.hide();
+    this.$repoSlugTypeahead.val('');
+    $('.typeahead__result, .typeahead__cancel-button').remove();
+    $('.typeahead__container').removeClass('cancel');
+
+    // initialise repo-typeahead for selected org
+    this.$repoSlugTypeahead.typeahead({
+      order: 'asc',
+      maxItem: 0,
+      source: {
+        data: repos,
+      },
+      callback: {
+        onClickAfter: (node) => {
+          const index = repos.indexOf(node[0].value);
+
+          this.getRepoDescription(descriptions[index]);
+          this.repoSelected = true;
+        },
+        onCancel: () => {
```

Comment:
Is this block conceptually doing the same thing done when you reset the typeahead component?

Reply:
👍